src/execution_plan/ops/op_index_scan.c

```diff
@@ -27,14 +30,14 @@ OpBase *NewIndexScanOp(Graph *g, Node *node, IndexIter *iter) {
   return (OpBase*)indexScan;
 }
 
-OpResult IndexScanConsume(OpBase *opBase, Record *r) {
+OpResult IndexScanConsume(OpBase *opBase, Record r) {
   IndexScan *op = (IndexScan*)opBase;
 
   EntityID *nodeId = IndexIter_Next(op->iter);
   if (!nodeId) return OP_DEPLETED;
 
-  Graph_GetNode(op->g, *nodeId, op->node);
-  Record_AddEntry(r, op->node->alias, SI_PtrVal(op->node));
+  Node *n = Record_GetNode(r, op->nodeRecIdx);
+  Graph_GetNode(op->g, *nodeId, n);
```

Comment:
Maybe add a comment here to indicate that `Graph_GetNode` is writing into the Record.

Reply:
👍